material-stepper/src/main/java/com/stepstone/stepper/type/TabsStepperType.java

```diff
@@ -57,11 +57,11 @@ public void onStepSelected(int newStepPosition) {
      * {@inheritDoc}
      */
     @Override
-    public void onNewAdapter(@NonNull AbstractStepAdapter<?> stepAdapter) {
+    public void onNewAdapter(@NonNull StepAdapter<?> stepAdapter) {
         List<Integer> titles = new ArrayList<>();
         final int stepCount = stepAdapter.getCount();
         for (int i = 0; i < stepCount; i++) {
-            final Step step = stepAdapter.getItem(i);
+            final Step step = (Step) stepAdapter.createStep(i);
```

Comment:
Now I get it, you're right. I was thinking about making public interface` StepAdapter<T extends Step>`, but then all the Frags example would break.
Do you have any ideas ? Gonna think about it for a time!